v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java

```diff
@@ -0,0 +1,188 @@
+    },
+    optionsClass = SourceDbToSpannerOptions.class,
+    flexContainerName = "source-db-to-spanner",
+    documentation =
+        "https://cloud.google.com/dataflow/docs/guides/templates/provided/sourcedb-to-spanner",
+    contactInformation = "https://cloud.google.com/support",
+    preview = true,
+    requirements = {
+      "The JDBC drivers for the relational database must be available.",
+      "The Spanner tables must exist before pipeline execution.",
+      "The Spanner tables must have a compatible schema.",
+      "The relational database must be accessible from the subnet where Dataflow runs."
+    })
+public class SourceDbToSpanner {
+
+  /**
+   * Main entry point for executing the pipeline. This will run the pipeline asynchronously. If
+   * blocking execution is required, use the {@link SourceDbToSpanner#run} method to start the
+   * pipeline and invoke {@code result.waitUntilFinish()} on the {@link PipelineResult}.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    UncaughtExceptionLogger.register();
+
+    // Parse the user options passed from the command-line
+    SourceDbToSpannerOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(SourceDbToSpannerOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Create the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  @VisibleForTesting
+  static PipelineResult run(SourceDbToSpannerOptions options) {
+    Pipeline pipeline = Pipeline.create(options);
+    Map<String, Set<String>> columnsToIgnore = getColumnsToIgnore(options);
+    Map<String, String> tableVsPartitionMap = getTablesVsPartitionColumn(options);
+    for (String table : getTablesVsPartitionColumn(options).keySet()) {
+      PCollection<Mutation> rows =
+          pipeline.apply(
+              "ReadPartitions_" + table,
+              getJdbcReader(
+                  table, tableVsPartitionMap.get(table), columnsToIgnore.get(table), options));
+      rows.apply("Write_" + table, getSpannerWrite(options));
+    }
+    return pipeline.run();
+  }
+
+  private static Map<String, String> getTablesVsPartitionColumn(SourceDbToSpannerOptions options) {
+    String[] tables = options.getTables().split(",");
+    String[] partitionColumns = options.getPartitionColumns().split(",");
+    if (tables.length != partitionColumns.length) {
+      throw new RuntimeException(
+          "invalid configuration. Partition column count does not match " + "tables count.");
+    }
+    Map<String, String> tableVsPartitionColumn = new HashMap();
+    for (int i = 0; i < tables.length; i++) {
+      tableVsPartitionColumn.put(tables[i], partitionColumns[i]);
+    }
+    return tableVsPartitionColumn;
+  }
+
+  private static Map<String, Set<String>> getColumnsToIgnore(SourceDbToSpannerOptions options) {
```

Comment:
unit testing scenarios here like ":column1" (empty table name), empty column name, trailing whitespaces etc.

But its fair to assume the user is not trying to break our code :P

Reply:
added todo here